src/twisted/logger/test/test_format.py

```diff
@@ -429,3 +430,94 @@ class Unformattable(object):
 
     def __repr__(self):
         return str(1 / 0)
+
+
+
+class CapturedError(Exception):
+    """
+    A captured error for use in format tests.
+    """
+
+
+
+class FormatEventWithTracebackTests(unittest.TestCase):
+    """
+    Tests for L{formatEventWithTraceback}, all of which ensure that the
+    returned type is a UTF-8.
+    """
+
+    def test_formatEventWithTraceback(self):
+        """
+        An event with a C{log_failure} key will have a traceback appended.
+        """
+        f = Failure(CapturedError("This is a fake error"))
+        event = {
+            "log_format": u"This is a test log message"
+        }
+        event["log_failure"] = f
+        eventText = formatEventWithTraceback(event)
+        self.assertIsInstance(eventText, unicode)
+        self.assertIn('Traceback', eventText)
```

Comment:
A better test would be:

```python
try:
    raise CapturedError("This is a fake error")
except CapturedError:
    f = Failure()
...
self.assertIn(f.getTraceback().decode('utf-8'), eventText)
...
```

That's because it would exercise the code with a real traceback.

Reply:
Makes sense.  What I submit will be slightly different as eventText is always unicode and getTraceback() is the native string type (the decode call fails on P3).